pkg/storage/stores/indexshipper/compactor/deletion/delete_request.go

```diff
@@ -133,23 +139,47 @@ func (d *DeleteRequest) IsDeleted(entry retention.ChunkEntry) (bool, []retention
 
 	intervals := make([]retention.IntervalFilter, 0, 2)
 
+	// chunk partially deleted from the end
 	if d.StartTime > entry.From {
+		// Add the deleted part with Filter func
+		if ff != nil {
+			intervals = append(intervals, retention.IntervalFilter{
+				Interval: model.Interval{
+					Start: d.StartTime,
+					End:   entry.Through,
+				},
+				Filter: ff,
+			})
+		}
```

Comment:
Delete request with a line filter that partially covers a chunk should have the deleted interval with a line filter.